modules/opensearch/includes/opensearch.search.inc

```diff
@@ -385,3 +382,38 @@ function opensearch_search_map_common_fields() {
     TingSearchCommonFields::SUBJECT => 'facet.subject',
   ];
 }
+
+
+/**
+ * Constructs a TingSearchRequest based on a "reference" search query.
+ *
+ * The implementation should assume that the user has entered a string with the
+ * express purpose of finding a specific material. That is, if the string seems
+ * to match the pattern of a unique material ID, the query should be constructed
+ * to look up by id.
+ *
+ * The provider can also choose to support advanced provider-specific querying.
+ *
+ * @param string $query_string
+ *   The user-provided search query.
+ *
+ * @return \Ting\Search\TingSearchRequest
+ *   The prepared query object.
+ */
```

Comment:
I like the level of commenting here. How about moving it to a reference file e.g. `search.api.php` within the `ting` module file for reference and change implementations to: `Implements hook_search_prepare_reference_query`

On the other hand I know that it is not exactly a Drupal hook implementation but a Ctools plugin so...

Reply:
My understanding was that that is basically the role of connie? That is, it should be an example of how a provider should handle its inputs and outputs.